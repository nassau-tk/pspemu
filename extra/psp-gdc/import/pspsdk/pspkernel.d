module pspsdk.pspkernel;

/*
#include <pspuser.h>
#include <pspiofilemgr_kernel.h>
#include <psploadcore.h>
#include <pspstdio_kernel.h>
#include <pspsysreg.h>
#include <pspkdebug.h>
#include <pspintrman_kernel.h>
#include <pspmodulemgr_kernel.h>
*/

/**
 * Set the $pc register to a kernel memory address.
 *
 * When the PSP's kernel library stubs are called, they expect to be accessed
 * from the kernel's address space.  Use this function to set $pc to the kernel
 * address space, before calling a kernel library stub.
 */
/*
#define pspKernelSetKernelPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"lui    $9, 0x8000\n\t" \
	"or     $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/

/**
 * Set the $pc register to a user memory address.
 *
 */
/*
#define pspKernelSetUserPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"li     $9, 0x7FFFFFFF\n\t" \
	"and    $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/