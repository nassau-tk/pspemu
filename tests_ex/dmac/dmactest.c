#include <pspkernel.h>
#include <oslib/oslib.h>

PSP_MODULE_INFO("dmac test", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER);

char dataSource[] = "Hello World. This is a test to check if sceDmacMemcpy works.";
char dataDest[sizeof(dataSource)] = {0};

int main(int argc, char *argv[]) {
	Kprintf("%s\n", dataSource);
	Kprintf("%s\n", dataDest);

	oslUncacheData(dataSource, sizeof(dataSource));
	oslUncacheData(dataDest, sizeof(dataDest));
	sceDmacMemcpy(dataDest, dataSource, sizeof(dataSource));

	Kprintf("%s\n", dataDest);

	void *ptr;
	ptr = memalign(128 , 2048); Kprintf("%d\n", ((int)ptr) % 128);
	ptr = memalign(1024, 2048); Kprintf("%d\n", ((int)ptr) % 1024);
	//ptr = memalign(100 , 2048); Kprintf("%d\n", ((int)ptr) % 100);

	//Kprintf("%i bytes available\n", oslGetRamStatus().maxAvailable);

	return 0;
}