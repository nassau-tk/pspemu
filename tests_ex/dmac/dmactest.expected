Hello World. This is a test to check if sceDmacMemcpy works.

Hello World. This is a test to check if sceDmacMemcpy works.
0
0